erpnext/hr/doctype/employee/employee.py

```diff
@@ -257,46 +257,24 @@ def send_birthday_reminders():
 		 	}
 		)
 		employee_emails = get_employee_emails(employee_list)
-		birthday_names = [name["employee_name"] for name in birthdays]
-		birthday_emails = [email["user_id"] or email["personal_email"] or email["company_email"] for email in birthdays]
-
-		birthdays.append({'company_email': '','employee_name': '','personal_email': '','user_id': ''})
-
-		for e in birthdays:
-			if e['company_email'] or e['personal_email'] or e['user_id']:
-				if len(birthday_names) == 1:
-					continue
-				recipients = e['company_email'] or e['personal_email'] or e['user_id']
-
-
+		
+		for birthday in birthdays:
+			if frappe.db.exists("Email Template", frappe.db.get_single_value("HR Settings", "birthday_email_template")):
+				email_template = frappe.get_doc("Email Template", frappe.db.get_single_value("HR Settings", "birthday_email_template")).response
 			else:
-				recipients = list(set(employee_emails) - set(birthday_emails))
+				email_template = "Happy Birthday {0}! \U0001F603".format(birthday["employee_name"])
 
-			frappe.sendmail(recipients=recipients,
-				subject=_("Birthday Reminder"),
-				message=get_birthday_reminder_message(e, birthday_names),
-				header=['Birthday Reminder', 'green'],
+			message = frappe.render_template(email_template, birthday)
+			frappe.sendmail(recipients=employee_emails,
+				subject=_("Happy Birthday"),
+				message=message
 			)
 
-def get_birthday_reminder_message(employee, employee_names):
-	"""Get employee birthday reminder message"""
-	pattern = "</Li><Br><Li>"
-	message = pattern.join(filter(lambda u: u not in (employee['employee_name']), employee_names))
-	message = message.title()
-
-	if pattern not in message:
-		message = "Today is {0}'s birthday \U0001F603".format(message)
-
-	else:
-		message = "Today your colleagues are celebrating their birthdays \U0001F382<br><ul><strong><li> " + message +"</li></strong></ul>"
-
-	return message
-
 
 def get_employees_who_are_born_today():
 	"""Get Employee properties whose birthday is today."""
 	return frappe.db.get_values("Employee",
-		fieldname=["name", "personal_email", "company", "company_email", "user_id", "employee_name"],
+		fieldname=["name", "personal_email", "company", "company_email", "user_id", "employee_name", "image", "date_of_birth"],
```

Comment:
Are these the fields that are available in the template? Show these in the HTML then.

Reply:
thought we don't need all those details to wish a person

Reply:
Actually, why are we filtering fields, all of them should be available. Change this to a `frappe.get_all` call. Something like:

```python
frappe.get_all("Employee", filters={filters}, fields=["*"])
```

That will get all the fields and just pass it to the template. Saves us from changing the code if the user wants to use more information.